### PR TITLE
First drop of react unit tests

### PR DESCRIPTION
**Pull request by neehar18**
In reference to: https://github.com/redhat-developer/vscode-granite/issues/49

### CODE REVIEW

**Comment by neehar18** [comment]
@fbricon the CI build failure is because there are conflicting type definitions between the installed versions of @types/mocha and @types/jest. 

Since I am using Jest for testing, can I remove @types/mocha from devDependencies?

File: package.json [hunk 1]
```diff
@@ -74,7 +79,29 @@
...
+      "ts",
+      "tsx",
+      "js",
+      "jsx",
+      "json",
+      "node"
+    ],
+    "moduleNameMapper": {
+      "\\.(css|less|scss|sass)$": "identity-obj-proxy"
+    }
```
Comment (fbricon) [inline package.json:104]: those changes seem unnecessary, since the tests only affect the webviews/ package

File: package.json [hunk 2]
```diff
@@ -61,9 +61,14 @@
     "pretest": "npm run compile-tests && npm run compile && npm run lint",
     "check-types": "tsc --noEmit",
     "lint": "eslint src --ext ts",
-    "test": "vscode-test"
+    "test": "vscode-test",
+    "test:jest": "jest",
+    "test:ci": "jest --ci --reporters=default --reporters=jest-junit"
```
Comment (fbricon) [inline package.json:66]: this should delegate the jest tests from the webviews package

File: webviews/src/StatusCheck.tsx [hunk 3]
```diff
@@ -1,5 +1,5 @@
 import { VscArrowCircleDown, VscCircleLarge, VscCircleLargeFilled, VscPass, VscPassFilled } from "react-icons/vsc";
-
+import React from 'react';
```
Comment (fbricon) [inline webviews/src/StatusCheck.tsx:2]: seems unnecessary

File: webviews/src/App.tsx [hunk 4]
```diff
@@ -6,6 +6,7 @@ import { ProgressData } from "../../src/commons/progressData";
 import { getStandardName } from "../../src/commons/naming";
 import { ModelStatus, ServerStatus } from "../../src/commons/statuses";
 import { StatusCheck, StatusValue } from "./StatusCheck";
+import React from 'react';
```
Comment (fbricon) [inline webviews/src/App.tsx:9]: seems unnecessary

**Review by fbricon — CHANGES_REQUESTED** [review]
You should not have a conflict between mocha and jest since the main package doesn't and shouldn't be concerned about react and jest

File: package.json [hunk 5]
```diff
@@ -61,7 +61,9 @@
     "pretest": "npm run compile-tests && npm run compile && npm run lint",
     "check-types": "tsc --noEmit",
     "lint": "eslint src --ext ts",
-    "test": "vscode-test"
+    "test": "vscode-test",
+    "test:jest": "cd webviews && npm run test",
+    "test:ci": "cd webviews && jest --ci --reporters=default --reporters=jest-junit"
```
Comment (fbricon) [inline package.json:66]: test:ci seems unused

File: webviews/src/ProgressBar.test.tsx [hunk 6]
```diff
@@ -2,7 +2,7 @@ import { render, screen } from '@testing-library/react';
 import ProgressBar from './ProgressBar';
 import { ProgressData } from '../../src/commons/progressData';
 import '@testing-library/jest-dom';
-import React from 'react';
+//import React from 'react';
```
Comment (fbricon) [inline webviews/src/ProgressBar.test.tsx:5]: don't comment, delete

**Comment by fbricon** [comment]
the build fails and there are conflicts to fix

**Comment by neehar18** [comment]
Closing the PR as this is addressed in https://github.com/redhat-developer/vscode-granite/pull/123

**Pull request closed without merging**